src/storage.rs

```diff
@@ -194,51 +204,29 @@ impl<'a> Builder {
     /// # Description
     /// Sets a string with work dir as pattern for blob naming.
     /// If path not exists, Storage will try to create at initialization stage.
-    /// # Panics
-    /// Panics if path is in readonly mode or path isn't a directory
     /// # Examples
     /// ```no-run
     /// let builder = Builder::new().work_dir("/tmp/pearl/");
     /// ```
     pub fn work_dir<S: Into<PathBuf>>(mut self, work_dir: S) -> Self {
         debug!("set work dir");
         let path: PathBuf = work_dir.into();
-        debug!("check path existence");
-        if !path.exists() {
-            warn!("no such file or directory: {}", path.display());
-        }
-        debug!("load metadata");
-        let metadata = path
-            .metadata()
-            .map_err(|e| error!("metadata load failed: {}", e))
-            .unwrap();
-        let permissions = metadata.permissions();
-        debug!("check is path readonly");
-        debug!("check is path a dir");
-        if permissions.readonly() {
-            error!("{} is in readonly mode", path.display());
-            panic!("work dir must have write permission");
-        } else if !metadata.is_dir() {
-            error!("{} is not a directory", path.display());
-            panic!("work dir must be a directory");
-        }
-        info!("work dir set: {}", path.display());
+        info!("work dir set to: {}", path.display());
         self.config.work_dir = Some(path);
         self
     }
 
     /// # Description
-    /// Sets storage max blob files num
+    /// Sets blob file max size
     /// Must be greater than zero
-    /// Limited by `u32::MAX`
-    pub fn max_blobs_num<U: Into<u32>>(mut self, max_blobs_num: U) -> Self {
-        let mbn = max_blobs_num.into();
-        debug!("cmp {} with 0", mbn);
-        if mbn > 0 {
-            self.config.max_blobs_num = Some(mbn);
-            info!("maximum blobs number set to: {}", mbn);
+    pub fn max_blob_size<U: Into<usize>>(mut self, max_blob_size: U) -> Self {
+        let mbs = max_blob_size.into();
+        debug!("cmp {} with 0", mbs);
+        if mbs > 0 {
+            self.config.max_blob_size = Some(mbs);
+            info!("maximum blob size set to: {}", mbs);
         } else {
-            error!("zero blobs num is useless, not set");
+            error!("zero size blobs is useless, not set");
```

Comment:
We can consider `0` as - use single blob file always (could be some use case)

Reply:
I am not agree. Zero size of the blob is definitely invalid value. For the unlimited size of the blob it is easier to set some large value for this field (obviously, there is no storage system that can contain file of size = 2^64 bytes)

Reply:
Left as is